test/fit_tests/README.md

```diff
@@ -0,0 +1,161 @@
+                          from global_config.json
+      -xunit              generates xUnit XML report files
+      -list               generates test list only
+      -sku SKU            node SKU, example:Phoenix, default=all
+      -obmmac OBMMAC      node OBM MAC address, example:00:1e:67:b1:d5:64
+      -nodeid NODEID      node identifier string of a discovered node, example:
+                          56ddcf9a8eff16614e79ec74
+      -http               forces the tests to utilize the http API protocol
+      -https              forces the tests to utilize the https API protocol
+      -port PORT          API port number override, default from
+                          global_config.json
+      -v V                Verbosity level of console output, default=0, Built Ins:
+                          0: No debug, 2: User script output, 4: rest calls and
+                          status info, 6: other common calls (ipmi, ssh), 9: all
+                          the rest
+
+This example will run the RackHD installer onto stack 1 via the wrapper script:
+
+    python run_tests.py -stack 1 -test deploy/run_rackhd_installer.py
+
+
+The -stack or -ora argument can be omitted when running on the server or appliance. The test defaults to localhost:8080 for API calls.
+
+This example will run the smoke test from the appliance node or the default Vagrant test bed:
+
+    python run_tests.py -test tests -group smoke
+
+
+Alternatively tests can be run directly from nose. Runtime parameters such as ORA address must be set in the environment.
+
+The following example will run all the entire test harness from a third party machine to ORA at 192.168.1.1:
+
+    export ORA=192.168.1.1
+    nosetests -s tests
+
+
+## Running individual tests
+
+Individual test scripts or tests may be executed using the following 'Nose' addressing scheme:
+
+    test_script_path:classname.testname
+
+
+For example, to run the test 'test_rackhd11_api_catalogs' in script 'tests/rackhd11/test_rackhd11_api_catalogs.py' on stack 1:
+
+    python run_tests.py -stack 11 -test tests/rackhd11/test_rackhd11_api_catalogs.py:rackhd11_api_catalogs.test_api_11_catalogs
+
+
+## Running FIT tests on Vagrant RackHD
+
+Install the Vagrant box using the instructions here:
+https://github.com/RackHD/RackHD/tree/master/example
+
+Install the Quanta simulator:
+
+    vagrant up quanta_d51
+
+Log into the Vagrant box:
+
+    vagrant ssh dev
+
+Clone FIT tests:
+
+    git clone https://github.com/onrack2/fit_tests
+
+Load virtual environment:
+
+    cd fit_tests/tests/fit_tests
+    virtualenv .venv
+    source .venv/bin/activate
+    pip install -r requirements.txt
+
+Load SKU packs:
+
+    python run_tests.py -test util/load_sku_packs.py
+
+Run Smoke test suite:
+
+    python run_tests.py -test tests/ -group smoke
+
+Use run_tests.py options if needed such as '-port' '-ora' '-https' to select IP address, port and protocol.
```

Comment:
Is the expectation that the quanta_d51 would have already gone through it's discovery and been processed entirely when the tests are invoked?

When I invoke it first, I get some failures, as happens when I invoke it after a discovery:

```
(.venv)vagrant@rackhd-demo:~/fit_tests/test/fit_tests$ python run_tests.py -test tests/ -group smoke
....F[
    "Nodeid: 57bcb91dc5b2bab107f6de91 Bad return code 404 for url: /redfish/v1/Chassis/57bcb91dc5b2bab107f6de91/Power",
    "Nodeid: QTFCJ05160195 Bad return code 404 for url: /redfish/v1/Chassis/QTFCJ05160195/Power"
]
F[
    "Nodeid: 57bcb91dc5b2bab107f6de91 Bad return code 404 for url: /redfish/v1/Chassis/57bcb91dc5b2bab107f6de91/Thermal",
    "Nodeid: QTFCJ05160195 Bad return code 404 for url: /redfish/v1/Chassis/QTFCJ05160195/Thermal"
]
F.................F...FFF................[
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller ID 57bcb91dc5b2bab107f6de8c, driveHealth failed to return any data",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller ID 57bcb91dc5b2bab107f6de8f, sel failed to return any data",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller ID 57bcb91dc5b2bab107f6de8e, selInformation failed to return any data",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller ID 57bcb91dc5b2bab107f6de8d, sdr failed to return any data",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller ID 57bcb91dc5b2bab107f6de90, chassis failed to return any data"
]
F[
    "Error: Poller 57bcb91dc5b2bab107f6de8c ID: driveHealth - Pollers not running",
    "Error: Poller 57bcb91dc5b2bab107f6de8f ID: sel - Pollers not running",
    "Error: Poller 57bcb91dc5b2bab107f6de8e ID: selInformation - Pollers not running",
    "Error: Poller 57bcb91dc5b2bab107f6de8d ID: sdr - Pollers not running",
    "Error: Poller 57bcb91dc5b2bab107f6de90 ID: chassis - Pollers not running"
]
F[
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8c Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8f Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8e Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8d Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de90 Failed to get current poller data, status 404"
]
F[
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8c Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8f Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8e Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8d Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de90 Failed to get current poller data, status 404"
]
F[
    "Node: 57bcb7f9f1b359b307c0e320 Poller: driveHealth 57bcb91dc5b2bab107f6de8c reported 2 failureCount",
    "Node: 57bcb7f9f1b359b307c0e320 Poller: sel 57bcb91dc5b2bab107f6de8f reported 2 failureCount",
    "Node: 57bcb7f9f1b359b307c0e320 Poller: selInformation 57bcb91dc5b2bab107f6de8e reported 2 failureCount",
    "Node: 57bcb7f9f1b359b307c0e320 Poller: sdr 57bcb91dc5b2bab107f6de8d reported 2 failureCount",
    "Node: 57bcb7f9f1b359b307c0e320 Poller: chassis 57bcb91dc5b2bab107f6de90 reported 3 failureCount"
]
F...[
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller ID 57bcb91dc5b2bab107f6de8c, driveHealth failed to return any data",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller ID 57bcb91dc5b2bab107f6de8f, sel failed to return any data",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller ID 57bcb91dc5b2bab107f6de8e, selInformation failed to return any data",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller ID 57bcb91dc5b2bab107f6de8d, sdr failed to return any data",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller ID 57bcb91dc5b2bab107f6de90, chassis failed to return any data"
]
F[
    "Error: Poller 57bcb91dc5b2bab107f6de8c ID: driveHealth - Pollers not running",
    "Error: Poller 57bcb91dc5b2bab107f6de8f ID: sel - Pollers not running",
    "Error: Poller 57bcb91dc5b2bab107f6de8e ID: selInformation - Pollers not running",
    "Error: Poller 57bcb91dc5b2bab107f6de8d ID: sdr - Pollers not running",
    "Error: Poller 57bcb91dc5b2bab107f6de90 ID: chassis - Pollers not running"
]
F[
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8c Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8f Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8e Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8d Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de90 Failed to get current poller data, status 404"
]
F[
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8c Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8f Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8e Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de8d Failed to get current poller data, status 404",
    "Error: Node 57bcb7f9f1b359b307c0e320 Poller_ID 57bcb91dc5b2bab107f6de90 Failed to get current poller data, status 404"
]
F[
    "Node: 57bcb7f9f1b359b307c0e320 Poller: driveHealth 57bcb91dc5b2bab107f6de8c reported 2 failureCount",
    "Node: 57bcb7f9f1b359b307c0e320 Poller: sel 57bcb91dc5b2bab107f6de8f reported 2 failureCount",
    "Node: 57bcb7f9f1b359b307c0e320 Poller: selInformation 57bcb91dc5b2bab107f6de8e reported 2 failureCount",
    "Node: 57bcb7f9f1b359b307c0e320 Poller: sdr 57bcb91dc5b2bab107f6de8d reported 2 failureCount",
    "Node: 57bcb7f9f1b359b307c0e320 Poller: chassis 57bcb91dc5b2bab107f6de90 reported 3 failureCount"
]
... (more failures after this) ...
```

End result:
```
Ran 166 tests in 289.672s

FAILED (SKIP=16, failures=33)
```

Are there any intended coordination points for specifically powering on additional machines and triggering a discovery process, or being aware of machines that are expected to "be there", what they should be, and how that should be configured for a test environment?

I did this all per the instructions for a Vagrant environment - I would expect all the tests could pass for anything we're proposing.